### PR TITLE
new flipBox function

### PR DESCRIPTION
**Pull request by jimbrig**
Addresses issue #70 

- add flip_box.R which contains the button functions and main flip_box function
- document (man/flip_box.Rd)
- added example app (inst/flip_box_demo)
- updated NAMESPACE

I also created a demo app [here](https://jimbrig2011.shinyapps.io/flip_box_demo/).

### CODE REVIEW

**Comment by jimbrig** [comment]
@DivadNojnarg apologies, forgot all about this!

I updated the code to include the buttons inside the body of the cards. Note the merge conflict in the DESCRIPTION due to your usage of roxygen's dev version `RoxygenNote: 7.1.1.9000`.

Feel free to change/comment on whatever as it is still quite limited as far as the interface goes.

I also attempted to bind the card as a shiny input and migrate the annoying JS from the function to an external JS file but had trouble getting the binding correct. I was able to simply assign a new id by calling:

```
function flipBox_js(id) {
  $("#" + id).on("click", ".btn-flip-front-id", function() {
    Shiny.setInputValue("fb_" + id, 0, { priority: "event"});
    $(this).tooltip("hide");
  });
  $("#" + id).on("click", ".btn-flip-back-id", function() {
    Shiny.setInputValue("fb_" + id, 1, { priority: "event"});
    $(this).tooltip("hide");
  });
}
```
and sourcing/calling the function from inside flip_box (as opposed to adding to deps.R). This was not the best solution however,  because observing the input did not work as expected without having to wrap the newly created input in a reactive/eventReactive, so I decided against it.

Since I did not think this hackish method was worthy of pushing, I just stashed it. I would like to attempt to fully create the binding as you did with bs4dash when I have some more time to spend on it cause my JS could use some work for sure (I've always avoided it when possible).

If you have any pointers let me know!

Thanks

File: NAMESPACE [hunk 1]
```diff
@@ -25,6 +25,10 @@ export(dropdownDivider)
 export(dropdownItem)
 export(dropdownItemList)
 export(flipBox)
```
Comment (DivadNojnarg) [inline NAMESPACE:27]: Here we used camelCase

File: NAMESPACE [hunk 2]
```diff
@@ -25,6 +25,10 @@ export(dropdownDivider)
...
+export(flip_box)
```
Comment (DivadNojnarg) [inline NAMESPACE:28]: Here it is snakeCase. User may get confused

**Review by DivadNojnarg — CHANGES_REQUESTED** [review]
Hi Jimmy,

thanks for your work. Looks better! Another request: I know that we are moving to the snake_case but shinydashboard and shinydashboardPlus are still using camelCase. Could you transform all functions in camelCase. This would make the package more consistent.

File: NAMESPACE (hunk 2 above)
Comment (jimbrig) [reply NAMESPACE]: right, the reasoning behind me keeping snake_case was so I did not overwrite your original flipBox code.

However, I went ahead and made your requested changes (better late than never!)

See [my latest commit](https://github.com/RinteRface/shinydashboardPlus/pull/77/commits/62a1af6ac560769082fa4b4d220de22994a84715).

Additionally, I added an input alias so the user can call input$< id >_value to access the card's state where 0=front, 1=back, and NULL = initial state. 

However, the card itself does not utilize a shiny input binding I simply used `Shiny.setInputValue` on a new id that adds a "_value" suffix to the assigned flipbox id.

This is displayed in the demo app nicely via `flipBoxDemo()`.

flipBoxDemo

File: NAMESPACE [hunk 3]
```diff
@@ -72,4 +75,7 @@ export(userPost)
 export(userPostMedia)
 export(userPostTagItem)
 export(userPostTagItems)
+importFrom(htmltools,tagList)
```
Comment (jimbrig) [inline NAMESPACE:78]: I will let you decide how you would like to handle the imports

File: R/flipBox.R [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+#' @importFrom shiny singleton
+flipBox <- function(id, 
+                    front_content, 
+                    back_content, 
+                    front_text = "More Info",
+                    back_text = "Back to Front") {
+  
+  if (is.null(id) || missing(id)) stop("card id cannot be null or missing!")
+  
+  alias <- paste0(id, "_value")
```
Comment (jimbrig) [inline R/flipBox.R:67]: new "alias" to access state of flipBox (could maybe use "_state" suffix instead)

File: inst/flipBoxDemo/app.R [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        )
+      )
+    )
+  ),
+  
+  server = function(input, output, session) {
+    
+    output$flipbox_value <- shiny::renderText({
+      req(input$id_value)
+      paste0("Flipbox Value: ", input$id_value)
```
Comment (jimbrig) [inline inst/flipBoxDemo/app.R:52]: displaying how to access value here in demo

**Review by jimbrig — COMMENTED** [review]
here's the requested changes + some enhancements previously discussed

**Review by jimbrig — COMMENTED** [review]
outdated now

**Comment by jimbrig** [comment]
@DivadNojnarg let me know if you need any assistance with the updated package/R CMD Checks.

**Pull request merged**